stakingModel/contracts/UBIScheme.sol

```diff
@@ -174,54 +151,58 @@ contract UBIScheme is AbstractUBI {
     }
 
     /* @dev Checks if the account has the privilege to claim. Only
-     * active accounts and new accounts that were been added the day
-     * before to the distribution list are eligible to claim.
+     * active accounts and accounts that were been added before to
+     * the pending list are eligible to claim. Adds inactive users
+     * to the pending list even if the user has not been fished.
      * @param account to update
      * @return A bool indicating if the account has the privilege to claim
      */
-    function _checkEntitlement(address account)
+    function _updateUserStatus(address account)
         private
         requireActive
-        onlyWhitelisted
         returns (bool)
     {
-        if(!checkActiveEntitlement(account)) { // the user is not active
-            if(checkNextDayEntitlement(account)) { // the user become to active
-                nextDayDistributionCount[currentDay] = nextDayDistributionCount[currentDay].sub(1);
-                activeUsersCount = activeUsersCount.add(1);
-                activeUsersAddresses[account] = true; // could not be fished
-                emit UserActivation(account, isNewUser(account)); // account and new or inactive
-            }
-            else { // the user has not last day entitlement
-                require(!nextDayDistributionAddresses[currentDay.add(1)][account].isEntitled, "has already added to the next day");
-                nextDayDistributionCount[currentDay.add(1)] = nextDayDistributionCount[currentDay.add(1)].add(1);
-                nextDayDistributionAddresses[currentDay.add(1)][account].isEntitled = true; // append the user to the next day distribution list
-                emit AddedToNextDay(account, isNewUser(account)); // account and new or inactive
+        if(isActiveUser(account)) {
+            require(!claimDay[currentDay].hasClaimed[account], "has already claimed");
+            return true;
+        }
+        if(pendingUsersAddresses[account].isPending) { // the user is in the pending list
+            if(pendingUsersAddresses[account].joinDay >= currentDay) { //already claimed today
```

Comment:
dont think joinday can be >= currentday. you probably want to check if this is an expired distribution account, ie for some reason distribution wasnt activate in the day he claimed. it means currentDay is > joinDay +1.
BUT I think we can remove this check, and anyways execute claim for the user, worse case he claimed by himself and the distribution claim will fail

Reply:
no longer pending list. distribute may trigger claim for that user but in case of the user already claimed, claim will emits a proper event for tracking but won't fail 
3a130df16ce6b41f40192c76921b58e63d2c5ccd